autoscale_cloudroast/test_repo/autoscale/system/integration/test_system_integration_rcv3.py

```diff
@@ -650,6 +665,116 @@ def test_scale_up_down_on_rcv3_and_clb(self):
+        pool_group = pool_group_resp.entity
+
+        # Construct a policy to scale up servers
+        server_count_before_autoscaling = self._get_node_counts_on_pool(
+            self.pool.id)['cloud_servers']
+
+        scale_amt = 2
+        policy_up_data = {'change': scale_amt, 'cooldown': 0}
+        expected_rcv3_server_count = (server_count_before_autoscaling
+                                      + scale_amt)
+        as_server_count = scale_amt
+
+        # Create the policy and execute it immediately
+        self.asb.create_policy_webhook(pool_group.id,
+                                       policy_up_data,
+                                       execute_policy=True)
+
+        server_list_after_up, nodes_after = self._verify_rcv3_group_state(
+            pool_group,
+            pool_group.launchConfiguration.loadBalancers[0].loadBalancerId,
+            as_server_count,
+            expected_rcv3_server_count,
+            as_server_count)
+
+        # Remove one of the servers from the LB pool
+        self.rcv3_client.remove_server_from_pool(self.pool.id,
+                                                 server_list_after_up[0])
+
+        post_del_servers, post_del_nodes = self._verify_rcv3_group_state(
+            pool_group,
+            pool_group.launchConfiguration.loadBalancers[0].loadBalancerId,
+            as_server_count,
+            expected_rcv3_server_count - 1,
+            as_server_count)
+
+        # Construct, create, and execute a policy to scale back to min.
+        policy_down_data = {'change': -scale_amt, 'cooldown': 0}
+        expected_rcv3_server_count = server_count_before_autoscaling
+        as_server_count = as_server_count - scale_amt
+
+        self.asb.create_policy_webhook(pool_group.id,
+                                       policy_down_data,
+                                       execute_policy=True)
+
+        self._verify_rcv3_group_state(
+            pool_group,
+            pool_group.launchConfiguration.loadBalancers[0].loadBalancerId,
+            as_server_count,
+            expected_rcv3_server_count,
+            as_server_count)
+
+        # Confirm that after scaling back to zero, the lb state matches the
+        # inital state
+        final_node_list = [n.id for n in self.rcv3_client.get_nodes_on_pool(
+            self.pool.id).entity.nodes]
+        self.assertEquals(set(initial_node_list), set(final_node_list),
+                          msg='The set of nodes on the pool is incorrect'
+                          'after scaling')
+
+    def _verify_rcv3_group_state(self, group, pool_id,
+                                 num_as, num_rc, num_nova):
+        """
+        Verify that the expected state of rcv3, autoscale, and nova is reached
+        """
+        server_list = []
+        # Since zero servers is an exit condtion that raises and exception in
+        # wait_for_expected_number_of_servers, if the expected number is zero
```

Comment:
Yeah, I'll take a look at fixing that in a later PR. For now, I wanted to touch a minimal number of files to finish the RCv3 tests.